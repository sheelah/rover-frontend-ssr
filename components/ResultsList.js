import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Loading from './Loading.js';
import ResultsListItem from './ResultsListItem';

class ResultsList extends Component {
  render() {
    const { results, isLoading, haveSearched } = this.props;

    if (isLoading) {
      return (
        <main className='search-results'>
          <div className='wrapper wrapper--icon'>
            <Loading />
          </div>
        </main>
      );
    }

    if (!isLoading && haveSearched && results.length  === 0) {
      // No matches were found
      return (
        <main className='search-results'>
          <div className='wrapper'>
            <p className="error--no-results">
              Sorry - no matches were found! Try changing your search options.
            </p>
          </div>
        </main>
      );
    }

    return (
      <main className='search-results'>
        <div className='wrapper'>
          <ul className='search-results__list'>
            {results.map(item => {
              return <ResultsListItem key={item.person_opk}
                person={item} />;
            })}
          </ul>
        </div>
      </main>
    );
  }
}

ResultsList.propTypes = {
  results: PropTypes.array.isRequired,
  isLoading: PropTypes.bool.isRequired,
  haveSearched: PropTypes.bool.isRequired
};

export default ResultsList;
