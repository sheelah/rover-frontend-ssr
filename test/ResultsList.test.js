import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ResultsList from '../components/ResultsList.js';

describe('No results', () => {
  const results = [];
  const haveSearched = false;
  const isLoading = true;

  it('Renders a loader when a search is active', () => {
    const component = shallow(
      <ResultsList
        results={results}
        isLoading={isLoading}
        haveSearched={haveSearched}
      />
    );
    expect(component.find('.search-results .wrapper--icon')).toHaveLength(1);
  });

  it('Renders correct results list markup', () => {
    const component = renderer.create(
      <ResultsList
          results={results}
          isLoading={isLoading}
          haveSearched={haveSearched}
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
