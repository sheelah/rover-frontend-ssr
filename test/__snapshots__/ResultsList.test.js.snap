// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`No results Renders correct results list markup 1`] = `
<main
  className="search-results"
>
  <div
    className="wrapper wrapper--icon"
  >
    <svg
      className="icon--loading"
      height="40"
      viewBox="0 0 40 40"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.201 5.169c-8.254 0-14.946 6.692-14.946 14.946 0 8.255 6.692 14.946 14.946 14.946s14.946-6.691 14.946-14.946c-.001-8.254-6.692-14.946-14.946-14.946zm0 26.58c-6.425 0-11.634-5.208-11.634-11.634 0-6.425 5.209-11.634 11.634-11.634 6.425 0 11.633 5.209 11.633 11.634 0 6.426-5.208 11.634-11.633 11.634z"
        opacity=".2"
      />
      <path
        d="M15.447 30.7l-1.235 3.07a14.855 14.855 0 0 0 7.675.942l-.466-3.28a11.57 11.57 0 0 1-5.974-.732z"
      >
        <animateTransform
          attributeName="transform"
          attributeType="xml"
          dur="0.7s"
          from="0 20 20"
          repeatCount="indefinite"
          to="360 20 20"
          type="rotate"
        />
      </path>
    </svg>
  </div>
</main>
`;
